### PR TITLE
login: Show `visiblePassword` type keyboard only when password is unobscured

### PR DESCRIPTION
**Pull request by rajveermalviya**
`keyboardType: TextInputType.visiblePassword` contradicts with `obscureText: true`, resulting in Gboard not obscuring the clipboard preview.

| `obscureText: true` | `obscureText: false` |
|-|-|
| Screenshot 2024-07-08 at 04 09 14 | Screenshot 2024-07-08 at 04 08 46 |

Fixes: #743

(This PR was result of pair-programming session with @sm-sayedi)

### CODE REVIEW

**Review by Khader-1 — APPROVED** [review]
Thanks @rajveermalviya! LGTM.

**Review by kenclary — APPROVED** [review]
lgtm

**Comment by chrisbobbe** [comment]
Cool, thanks, and I'm glad you had a chance to pair-program with @sm-sayedi for this :)

LGTM, over to Greg for his review.

**Comment by gnprice** [comment]
Thanks @rajveermalviya and @sm-sayedi for the fix!

Looks good — merging. I found the commit message a bit hard to parse, so I expanded the text somewhat and opened it with some framing of the context:

```diff
-    login: Show `visiblePassword` type keyboard only when password is unobscured
+    login: Show `visiblePassword` keyboard only when password is visible
 
-    `keyboardType: TextInputType.visiblePassword` contradicts with
-    `obscureText: true`, resulting in Gboard not obscuring the
-    clipboard preview.
+    When Gboard offers a suggestion from the clipboard, the suggestion
+    will be obscured given `obscureText: true`... except that if we also
+    pass `keyboardType: TextInputType.visiblePassword`, then that causes
+    the suggestion's contents to be visible after all.  Avoid that.
 
     Fixes: #743
-
     Co-authored-by: Sayed Mahmood Sayedi <smahmood.sayedi@gmail.com>
```

Also joined the two "Foo: bar" lines into one stanza, as that's the convention for that sort of line in a Git commit message.
